### PR TITLE
feat(newserver) Arma Reforger

### PR DESCRIPTION
**Pull request by FliesWithWind**
# Description

This PR adds Arma Reforger support. I hope I did not miss anything. 

There is one issue with Arma Reforger server that I am aware of - libssl1.1 is not available on Ubuntu 22.04 (not sure about other distros). It can be installed in other ways ( [example](https://askubuntu.com/questions/1403619/mongodb-install-fails-on-ubuntu-22-04-depends-on-libssl1-1-but-it-is-not-insta)), but I am not sure what is the best way to handle this through LinuxGSM. 

Fixes #[issue]

## Type of change

* [ ] Bug fix (a change which fixes an issue).
* [ ] New feature (change which adds functionality).
* [x] New Server (new server added).
* [ ] Refactor (restructures existing code).
* [ ] Comment update (typo, spelling, explanation, examples, etc).

## Checklist

PR will not be merged until all steps are complete.

* [ ] This pull request links to an issue.
* [ ] This pull request uses the `develop` branch as its base.
* [ ] This pull request Subject follows the Conventional Commits standard.
* [x] This code follows the style guidelines of this project.
* [x] I have performed a self-review of my code.
* [ ] I have checked that this code is commented where required.
* [x] I have provided a detailed with enough description of this PR.
* [ ] I have checked If documentation needs updating.

## Documentation

If documentation does need updating either update it by creating a PR (preferred) or request a documentation update.
* User docs: https://github.com/GameServerManagers/LinuxGSM-Docs
* Dev docs: https://github.com/GameServerManagers/LinuxGSM-Dev-Docs

**Thank you for your Pull Request!**

### CODE REVIEW

**Comment by Marcin648** [comment]
I not decision-making guy in LGSM but,
it is allready done in https://github.com/h3o66/LinuxGSM/tree/feature/armar and https://github.com/Marcin648/LinuxGSM/tree/feature/armar
`-backendlog -nothrow -gproj` is not required in my opinion.
`engine="realvirtuality"` should be `enfusion`,
and for json config file better is to use jq insted of regex.
@h3o66 I think we should to PR to your repo. :)

File: lgsm/config-default/config-lgsm/armareforgerserver/_default.cfg [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+# common.cfg - applies settings to every instance.
+# [instance].cfg - applies settings to a specific instance.
+
+#### Game Server Settings ####
+
+## Predefined Parameters | https://docs.linuxgsm.com/configuration/start-parameters
+ip="0.0.0.0"
+
+## Server Parameters | https://docs.linuxgsm.com/configuration/start-parameters#additional-parameters
+startparameters="-config ${servercfgfullpath} -backendlog -nothrow -profile ${serverprofile} -gproj ${gproj}"
```
Comment (h3o66) [inline lgsm/config-default/config-lgsm/armareforgerserver/_default.cfg:15]: Looks like this is taken from the manual:
https://community.bistudio.com/wiki/Arma_Reforger:Server_Hosting
Do we really need here all ?

File: lgsm/data/serverlist.csv [hunk 2]
```diff
@@ -3,6 +3,7 @@ ahl,ahlserver,Action Half-Life
 ahl2,ahl2server,Action: Source
 ark,arkserver,ARK: Survival Evolved
 arma3,arma3server,ARMA 3
+armareforger,armareforgerserver,ARMA Reforger
```
Comment (h3o66) [inline lgsm/data/serverlist.csv:6]: Not sure about this.
@dgibbs64 do you think the full name should be used here ?

File: lgsm/functions/info_game.sh [hunk 3]
```diff
@@ -93,6 +93,31 @@ fn_info_game_arma3(){
...
 
+fn_info_game_armareforger(){
+	# Config
+	if [ ! -f "${servercfgfullpath}" ]; then
+		servername="${unavailable}"
+		adminpassword="${unavailable}"
+		serverpassword="${unavailable}"
+		maxplayers="${zero}"
+	else
+		servername=$(grep "name" "${servercfgfullpath}" | sed -e 's/^[ \t]*//g' -e '/^\//d' -e 's/name//g' | tr -d '=\";,:' | sed -e 's/^[ \t]*//' -e 's/[ \t]*$//')
```
Comment (h3o66) [inline lgsm/functions/info_game.sh:104]: Like @Marcin648 i would prefer here the parsing via jq - this is something I would not do anymore for configparsing.
Note: also the configip should be added as well.
See this parsing as example this:
https://github.com/h3o66/LinuxGSM/commit/0757585ea3b8a8361e62895fa265f4a7b43f0f55#diff-670814a6c9341dd77e070ca74832863054f3e3998d5a339606b744e7bb3480f9

**Review by h3o66 — CHANGES_REQUESTED** [review]

**Comment by h3o66** [comment]
Also the PR should be based on develop and not on the master branch

**Comment by FliesWithWind** [comment]
@Marcin648 @h3o66 Thanks for the review and comments. I'll address them shortly.

I wasn't aware you guys were already working on it. We can move this PR to one of your repos if it will make it all easier in the end. ;)

**Comment by h3o66** [comment]
> @Marcin648 @h3o66 Thanks for the review and comments. I'll address them shortly.
> 
> I wasn't aware you guys were already working on it. We can move this PR to one of your repos if it will make it all easier in the end. ;)

Probably my fault, did not create a Draft PR after I pushed my changes.
On my Branch for example the dependencys are missing, see as well #3871

**Comment by FliesWithWind** [comment]
Alright then I will close this PR and drop some suggestions to your PR.

**Pull request closed without merging**